### PR TITLE
/Rk command for rop queries

### LINKED ISSUE #5396: Add /Rk command to query the rop/ namespace
/Rk should be an alias or a way to access the info of "k rop." 

Also, in addition, we want this list to be exported in JSON format and r2 commands using the /Rk* and /Rkj, also add /Rkq to list all the offsets of the rop gadgets like its done in aflq and such

### PR DESCRIPTION
**Pull request by Darredevil**
Fix #5396

### CODE REVIEW

File: libr/core/cmd_search.c [hunk 1]
```diff
@@ -1830,6 +1829,52 @@ static void do_string_search(RCore *core, struct search_parameters *param) {
...
+		return;
+	}
+
+	switch (*input) {
+		case 'q':
+			sdb_list = sdb_foreach_list (db_rop);
+			ls_foreach (sdb_list, sdb_iter, kv) {
+				r_cons_printf ("%s ", kv->key);
+			}
+			r_cons_printf("\n");
```
Comment (radare) [inline libr/core/cmd_search.c]: use r_cons_newline()

File: libr/core/cmd_search.c [hunk 2]
```diff
@@ -1830,6 +1829,52 @@ static void do_string_search(RCore *core, struct search_parameters *param) {
...
+			break;
+		case 'j':
+			r_cons_printf ("{\"gadgets\":[");
+			sdb_list = sdb_foreach_list (db_rop);
+			ls_foreach (sdb_list, sdb_iter, kv) {
+				if (json_first) {
+					json_first = false;
+				} else {
+					r_cons_printf (",");
```
Comment (radare) [inline libr/core/cmd_search.c]: use r_cons_print() if no format is involved

File: libr/core/cmd_search.c [hunk 3]
```diff
@@ -1830,6 +1829,52 @@ static void do_string_search(RCore *core, struct search_parameters *param) {
...
+	switch (*input) {
+		case 'q':
+			sdb_list = sdb_foreach_list (db_rop);
+			ls_foreach (sdb_list, sdb_iter, kv) {
+				r_cons_printf ("%s ", kv->key);
+			}
+			r_cons_printf("\n");
+			break;
+		case 'j':
+			r_cons_printf ("{\"gadgets\":[");
```
Comment (radare) [inline libr/core/cmd_search.c]: r_cons_print. not format

File: libr/core/cmd_search.c [hunk 4]
```diff
@@ -1830,6 +1829,52 @@ static void do_string_search(RCore *core, struct search_parameters *param) {
...
+	char *out;
+
+	db_rop = sdb_ns (core->sdb, "rop", false);
+	if (!db_rop) {
+		eprintf ("Error: could not find SDB 'rop' namespace\n");
+		return;
+	}
+
+	switch (*input) {
+		case 'q':
```
Comment (radare) [inline libr/core/cmd_search.c]: indent cases at the same level of the switch

File: libr/core/cmd_search.c [hunk 5]
```diff
@@ -1986,10 +2031,23 @@ static int cmd_search(void *data, const char *input) {
...
+			if (input[2] == '?') {
+				const char* help_msg[] = {
+					"Usage: /Rk", "", "Query stored ROP gadgets",
+					"/Rk", "", "Show gadgets",
+					"/Rkj", "", "JSON output",
+					"/Rkq", "", "List Gadgets offsets",
+					NULL};
+				r_core_cmd_help (core, help_msg);
+			} else {
+				rop_kuery (core, input+2);
```
Comment (radare) [inline libr/core/cmd_search.c]: add spaces around +

File: libr/core/cmd_search.c [hunk 6]
```diff
@@ -1830,6 +1829,52 @@ static void do_string_search(RCore *core, struct search_parameters *param) {
...
+			sdb_list = sdb_foreach_list (db_rop);
+			ls_foreach (sdb_list, sdb_iter, kv) {
+				if (json_first) {
+					json_first = false;
+				} else {
+					r_cons_printf (",");
+				}
+				r_cons_printf ("{\"address\":%s,\"size\":%s}", kv->key, kv->value);
+			}
+			r_cons_printf("]}\n");
```
Comment (radare) [inline libr/core/cmd_search.c]: missing space before (

File: libr/core/cmd_search.c [hunk 7]
```diff
@@ -1830,6 +1829,52 @@ static void do_string_search(RCore *core, struct search_parameters *param) {
...
+static void rop_kuery(void *data, const char *input) {
+	RCore *core = (RCore *)data;
+	Sdb *db_rop = NULL;
+	bool json_first = true;
+	SdbListIter *sdb_iter;
+	SdbList *sdb_list;
+	SdbKv *kv;
+	char *out;
+
+	db_rop = sdb_ns (core->sdb, "rop", false);
```
Comment (radare) [inline libr/core/cmd_search.c]: define and assign in the same line here `Sdb *db_rop = ...`

**Comment by Darredevil** [comment]
@radare fixed comments

**Comment by radare** [comment]
lgtm!

**Pull request merged**